uni-stark/src/prover.rs

```diff
@@ -47,9 +47,35 @@ where
     // Count the number of constraints that we have.
     let constraint_count = symbolic_constraints.len();
 
-    // Find the total degree of the multivariate constraint polynomial `C`.
+    // Each constraint polynomial looks like `C_j(X_1, ..., X_w, Y_1, ..., Y_w, Z_1, ..., Z_j)`.
+    // When evaluated on a given row, the X_i's will be the `i`'th element of the that row, the
+    // Y_i's will be the `i`'th element of the next row and the Z_i's correspond to evaluations of
+    // selector polynomials.
     //
-    // For now in comments we assume that `deg(C) = 3`.
+    // When we convert to working with trace polynomials, the `X_i`'s and `Y_i`'s will be replaced by
+    // the degree `N - 1` polynomials `T_i(x)` and `T_i(hx)` respectively. The selectors are a little
+    // more complicated however.
+    //
+    // In our our case, the selector polynomials are `S_1(x) = is_first_row`, `S_2(x) = is_last_row`
+    // and `S_3(x) = is_transition`. Both `S_1(x)` and `S_2(x)` are to polynomials of degree `N - 1`
+    // as they must be non zero only at a single location in the initial domain. However, `is_transition`
+    // is a polynomial of degree `1` as it simply need to be `0` on the last row.
+    //
+    // The constraint degree (`deg(C)`) is the linear factor of `N` in the constraint polynomial. In other
+    // words, it is roughly the total degree of `C` however, we treat `Z_3` as a constant term which does
+    // not contribute to the degree.
+    //
+    // E.g. C_j = Z_1(X_1^3 - X_2 * X_3 * X_4) would have degree `4`.
+    //      C_j = Z_3(X_1^3 - X_2 * X_3 * X_4) would have degree `3`.
```

Comment:
```suggestion
    // E.g. C_j = S_1 * (X_1^3 - X_2 * X_3 * X_4) would have degree `4`.
    //      C_j = S_3 * (X_1^3 - X_2 * X_3 * X_4) would have degree `3`.
```

Reply:
Here I do think the * is helpful, because at first I thought Z_i were functions.